### PR TITLE
Add API for setting visibility of crop margins, render crop margins as boxes

### PR DESCRIPTION
**Pull request by seanzhang15**
This PR adds a command to selectively visualize crop container margin and improves visualization by using colored box instead of line meshes to better visualize the remain volume

image

### CODE REVIEW

File: plugins/qtosgrave/qtosgviewer.cpp [hunk 1]
```diff
@@ -1999,6 +2017,11 @@ void QtOSGViewer::_MoveCameraZoom(float factor, bool isPan, float panDelta)
     _posgWidget->MoveCameraZoom(factor, isPan, panDelta);
 }
 
+void QtOSGViewer::_SetCropMarginVisible(const std::string& bodyName, const std::string& linkName, const std::string& geometryName, const std::string& cropContainerType, bool visible)
+{
+    _posgWidget->GetItemFromName(bodyName)->SetCropMarginVisible(linkName, geometryName, cropContainerType, visible);
```
Comment (braineo) [inline plugins/qtosgrave/qtosgviewer.cpp:2022]: check for nullptr

File: plugins/qtosgrave/osgrenderitem.cpp [hunk 2]
```diff
@@ -484,21 +485,26 @@ void KinBodyItem::Load()
...
-                        orgeom->GetPositiveCropContainerEmptyMargins(), 
-                        orgeom->GetDiffuseColor() + (-orgeom->GetDiffuseColor() + RaveVector<float>(1, 1, 1)) * 0.5 // tint of the geometry color
-                    );
+                    std::pair<std::string, std::string> linkGeometryNames(porlink->GetName(), orgeom->GetName());
+                    if (_cropMarginVisible.count(linkGeometryNames) != 0) {
+                        DrawCropContainerMargins(
+                            pgeometrydata, 
+                            orgeom->GetContainerInnerExtents(), 
+                            orgeom->GetNegativeCropContainerMargins(), 
+                            orgeom->GetPositiveCropContainerMargins(),
```
Comment (braineo) [inline plugins/qtosgrave/osgrenderitem.cpp:494]: let's remove the trailing whitespace

**Review by braineo — CHANGES_REQUESTED** [review]

**Review by braineo — APPROVED** [review]

**Comment by seanzhang15** [comment]
closing in favor of https://github.com/rdiankov/openrave/pull/1251

**Comment by cielavenir** [comment]
so you switched to rdiankov:crop-margins rather than your fork, thanks @seanzhang15

**Pull request closed without merging**